OtterLib/core/include/Factory.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+** EPITECH PROJECT, 2022
+** indie
+** File description:
+** Factory
+*/
+
+#ifndef FACTORY_HPP_
+#define FACTORY_HPP_
+
+#include "EntityManager.hpp"
+#include "Orchestrator.hpp"
+
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <functional>
+#include <map>
+
+namespace Otter::Core {
+    namespace pt = boost::property_tree;
+
+    class Component {};
+
+    class Factory {
```

Comment:
Add a doc for methods pls,

!! TAKE into accout that we don't see throw in prototype, so describe behavior correctly